File: app/src/main/java/com/example/producttest/database/DatabaseHelper.java
package com.example.producttest.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.database.sqlite.SQLiteStatement;
import android.util.Log;

import androidx.annotation.Nullable;

import com.example.producttest.model.Product;

import java.util.ArrayList;
import java.util.List;

public class DatabaseHelper extends SQLiteOpenHelper {
    private final String PRODUCT = "PRODUCT";
    public static final String COLUMN_ID = "id";
    public static final String COLUMN_NAME = "name";
    public static final String COLUMN_PRICE = "price";
    public static final String COLUMN_DES = "des";
    public static final String COLUMN_COUNT = "count";

    public static final String COLUMN_image = "image";
    private Context mContext;
    private static final String tablesqlite = "Product.sqlite";
    private SQLiteDatabase mDatabase;

    public DatabaseHelper(@Nullable Context context, @Nullable String name, @Nullable SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }
    public DatabaseHelper(Context context) {
        super(context, tablesqlite, null, 1);
        this.mContext = context;
        this.mDatabase = mDatabase;

    }


    public void removeAll() {
        SQLiteDatabase db = this.getWritableDatabase(); // helper is object extends SQLiteOpenHelper
        db.delete(PRODUCT, null, null);
    }
//    public void insertData (String name,String price,String des, byte [] image){
//        SQLiteDatabase database = getWritableDatabase();
//        String sql = "INSERT INTO "+PRODUCT+" VALUES (NULL,?,?,?,?)";
//        SQLiteStatement sqLiteStatement = database.compileStatement(sql);
//        sqLiteStatement.clearBindings();
//        sqLiteStatement.bindString(1,name);
//        sqLiteStatement.bindString(2,price);
//        sqLiteStatement.bindString(3,des);
//        sqLiteStatement.bindBlob(4,image);
//        sqLiteStatement.executeInsert();
//
//    }
    public long insertProduct(Product product){
        SQLiteDatabase sqLiteDatabase = this.getWritableDatabase();
        ContentValues contentValues = new ContentValues();
        contentValues.put(COLUMN_NAME,product.getName());
        contentValues.put(COLUMN_PRICE,product.getPrice());
        contentValues.put(COLUMN_COUNT,product.getCount());
        contentValues.put(COLUMN_DES,product.getDes());
//        contentValues.put(COLUMN_image,product.getImage());
        long result = sqLiteDatabase.insert(PRODUCT,null,contentValues);
        return result;
    }
//    public long updateProduct(Product product){
//        SQLiteDatabase sqLiteDatabase = this.getWritableDatabase();
//        ContentValues contentValues = new ContentValues();
//        contentValues.put(COLUMN_ID,product.getId());
//        contentValues.put(COLUMN_NAME,product.getName());
//        contentValues.put(COLUMN_PRICE,product.getPrice());
//        contentValues.put(COLUMN_DES,product.getDes());
//        long result = sqLiteDatabase.update(PRODUCT,contentValues,
//                COLUMN_ID+"=?",new String[]{product.getId()});
//        return result;
//    }
    public void updateProducts(Product product){
        SQLiteDatabase sqLiteDatabase = this.getWritableDatabase();
        ContentValues contentValues = new ContentValues();
//        contentValues.put(COLUMN_ID,product.getId());
        contentValues.put(COLUMN_NAME,product.getName());
        contentValues.put(COLUMN_PRICE,product.getPrice());
        contentValues.put(COLUMN_COUNT,product.getCount());
        contentValues.put(COLUMN_DES,product.getDes());
       sqLiteDatabase.update(PRODUCT,contentValues,COLUMN_ID+"=?",new String[]{product.id});

    }
    public long deleteProduct(String id){
        SQLiteDatabase sqLiteDatabase = this.getWritableDatabase();
        return sqLiteDatabase.delete(PRODUCT,COLUMN_ID+"=?",new String[]{id});
    }
    public List<Product> getAllProduct(){
        List<Product> productArrayList = new ArrayList<>();
        String SELECT = "SELECT * FROM " + PRODUCT;
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor cursor = db.rawQuery(SELECT,null);
        if (cursor.moveToFirst()){
            do {
                Product product = new Product();
                product.setId(cursor.getString(0));
                product.setName(cursor.getString(1));
                product.setPrice(cursor.getInt(2));
                product.setCount(cursor.getInt(3));
                product.setDes(cursor.getString(4));
//                product.setImage(cursor.getBlob(4));
                productArrayList.add(product);
            }while (cursor.moveToNext());
        }
        db.close();
        return productArrayList;
    }
    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase) {
        String CREATE_TABLE =
                "CREATE TABLE "+PRODUCT+"("+COLUMN_ID+
                        " INTEGER PRIMARY KEY," +
                        "" + COLUMN_NAME+" VARCHAR," +
                        "" + COLUMN_PRICE+" INTEGER," +
                        "" + COLUMN_COUNT+" INTEGER," +
                        "" + COLUMN_DES+" VARCHAR," +
                        "" + COLUMN_image+" BLOG)";
        sqLiteDatabase.execSQL(CREATE_TABLE);
        Log.d("CREATE_TABLE",CREATE_TABLE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {

    }
}

File: app/src/main/java/com/example/producttest/database/DatabaseHelper.kt
package com.example.producttest.database

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteDatabase.CursorFactory
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log
import com.example.producttest.model.Product
import java.util.*

class DatabaseHelper : SQLiteOpenHelper {
    private val PRODUCT = "PRODUCT"
    private var mContext: Context? = null
    private var mDatabase: SQLiteDatabase? = null

    constructor(context: Context?, name: String?, factory: CursorFactory?, version: Int) : super(context, name, factory, version) {}
    constructor(context: Context?) : super(context, tablesqlite, null, 1) {
        mContext = context
        mDatabase = mDatabase
    }

    fun removeAll() {
        val db = this.writableDatabase // helper is object extends SQLiteOpenHelper
        db.delete(PRODUCT, null, null)
    }

    //    public void insertData (String name,String price,String des, byte [] image){
    //        SQLiteDatabase database = getWritableDatabase();
    //        String sql = "INSERT INTO "+PRODUCT+" VALUES (NULL,?,?,?,?)";
    //        SQLiteStatement sqLiteStatement = database.compileStatement(sql);
    //        sqLiteStatement.clearBindings();
    //        sqLiteStatement.bindString(1,name);
    //        sqLiteStatement.bindString(2,price);
    //        sqLiteStatement.bindString(3,des);
    //        sqLiteStatement.bindBlob(4,image);
    //        sqLiteStatement.executeInsert();
    //
    //    }
    fun insertProduct(product: Product): Long {
        val sqLiteDatabase = this.writableDatabase
        val contentValues = ContentValues()
        contentValues.put(COLUMN_NAME, product.name)
        contentValues.put(COLUMN_PRICE, product.price)
        contentValues.put(COLUMN_COUNT, product.count)
        contentValues.put(COLUMN_DES, product.des)
        //        contentValues.put(COLUMN_image,product.getImage());
        return sqLiteDatabase.insert(PRODUCT, null, contentValues)
    }

    //    public long updateProduct(Product product){
    //        SQLiteDatabase sqLiteDatabase = this.getWritableDatabase();
    //        ContentValues contentValues = new ContentValues();
    //        contentValues.put(COLUMN_ID,product.getId());
    //        contentValues.put(COLUMN_NAME,product.getName());
    //        contentValues.put(COLUMN_PRICE,product.getPrice());
    //        contentValues.put(COLUMN_DES,product.getDes());
    //        long result = sqLiteDatabase.update(PRODUCT,contentValues,
    //                COLUMN_ID+"=?",new String[]{product.getId()});
    //        return result;
    //    }
    fun updateProducts(product: Product) {
        val sqLiteDatabase = this.writableDatabase
        val contentValues = ContentValues()
        //        contentValues.put(COLUMN_ID,product.getId());
        contentValues.put(COLUMN_NAME, product.name)
        contentValues.put(COLUMN_PRICE, product.price)
        contentValues.put(COLUMN_COUNT, product.count)
        contentValues.put(COLUMN_DES, product.des)
        sqLiteDatabase.update(PRODUCT, contentValues, COLUMN_ID + "=?", arrayOf(product.id))
    }

    fun deleteProduct(id: String): Long {
        val sqLiteDatabase = this.writableDatabase
        return sqLiteDatabase.delete(PRODUCT, COLUMN_ID + "=?", arrayOf(id)).toLong()
    }

    //                product.setImage(cursor.getBlob(4));
    val allProduct: MutableList<Product>
        get() {
            val productArrayList: MutableList<Product> = ArrayList()
            val SELECT = "SELECT * FROM $PRODUCT"
            val db = this.readableDatabase
            val cursor = db.rawQuery(SELECT, null)
            if (cursor.moveToFirst()) {
                do {
                    val product = Product()
                    product.id = cursor.getString(0)
                    product.name = cursor.getString(1)
                    product.price = cursor.getInt(2)
                    product.count = cursor.getInt(3)
                    product.des = cursor.getString(4)
                    //                product.setImage(cursor.getBlob(4));
                    productArrayList.add(product)
                } while (cursor.moveToNext())
            }
            db.close()
            return productArrayList
        }

    override fun onCreate(sqLiteDatabase: SQLiteDatabase) {
        val CREATE_TABLE = "CREATE TABLE " + PRODUCT + "(" + COLUMN_ID +
                " INTEGER PRIMARY KEY," +
                "" + COLUMN_NAME + " VARCHAR," +
                "" + COLUMN_PRICE + " INTEGER," +
                "" + COLUMN_COUNT + " INTEGER," +
                "" + COLUMN_DES + " VARCHAR," +
                "" + COLUMN_image + " BLOG)"
        sqLiteDatabase.execSQL(CREATE_TABLE)
        Log.d("CREATE_TABLE", CREATE_TABLE)
    }

    override fun onUpgrade(sqLiteDatabase: SQLiteDatabase, i: Int, i1: Int) {}

    companion object {
        const val COLUMN_ID = "id"
        const val COLUMN_NAME = "name"
        const val COLUMN_PRICE = "price"
        const val COLUMN_DES = "des"
        const val COLUMN_COUNT = "count"
        const val COLUMN_image = "image"
        private const val tablesqlite = "Product.sqlite"
    }
}